test/stringify.test.ts

```diff
@@ -170,53 +189,90 @@ a = 2
 
 ["testkey@"]
 a = 3
-`.trim()
+`.trim();
 
-	expect(stringify({ 'test-key123_': { a: 1 }, 'test key 123': { a: 2 }, 'testkey@': { a: 3 } }).trim()).toBe(expected)
-})
+  expect(
+    stringify({
+      'test-key123_': { a: 1 },
+      'test key 123': { a: 2 },
+      'testkey@': { a: 3 },
+    }).trim(),
+  ).toBe(expected);
+});
 
 it('does not produce invalid strings', () => {
-	const testObj = {
-		str1: 'test\n',
-		str2: 'test\x00',
-		str3: 'test"',
-		str4: 'test\\',
-		str5: 'test\x7f',
-	}
-
-	const stringified = stringify(testObj)
-	expect(stringified).not.toContain('\n"')
-	expect(stringified).not.toContain('\x00')
-	expect(stringified).toContain('\\"')
-	expect(stringified).toContain('\\\\')
-	expect(stringified).not.toContain('\x7f')
-})
+  const testObj = {
+    str1: 'test\n',
+    str2: 'test\x00',
+    str3: 'test"',
+    str4: 'test\\',
+    str5: 'test\x7f',
+  };
+
+  const stringified = stringify(testObj);
+  expect(stringified).not.toContain('\x00');
+  expect(stringified).toContain('\\"');
+  expect(stringified).toContain('\\\\');
+  expect(stringified).not.toContain('\x7f');
+});
+
+it('formats strings with newlines as multi-line literal strings', () => {
+  const testObj = {
+    multilineString: 'This string\nhas multiple\nlines in it',
+  };
+
+  const stringified = stringify(testObj);
+  expect(stringified).toContain("multilineString = '''");
+  expect(stringified).toContain('This string\nhas multiple\nlines in it');
+  expect(stringified).toContain("'''");
+});
+
+it('formats long strings as multi-line literal strings with line breaks', () => {
+  const testObj = {
+    longString:
+      'This is a very long string that should be formatted as a multi-line literal string because it exceeds the 120 character limit we have set in our formatting function. It should break nicely at spaces.',
+  };
+
+  const stringified = stringify(testObj);
+  expect(stringified).toContain("longString = '''");
+  expect(stringified).toContain("'''");
+  expect(stringified.split('\n').length).toBeGreaterThan(3); // At least 3 lines (opening quote, content with breaks, closing quote)
+});
```

Comment:
While I get the intent, the presence of more than 1 newline at the beginning of the string should be a failure.

```ts
expect(parse(stringify(testObj)).toStrictEqual(testObj)
```